pinot-spi/src/main/java/org/apache/pinot/spi/utils/IngestionConfigUtils.java

```diff
@@ -124,6 +126,25 @@ public static int getStreamConfigIndexFromPinotPartitionId(int partitionId) {
     return partitionId / PARTITION_PADDING_OFFSET;
   }
 
+  /**
+   * Getting the StreamConfig from StreamConfigs list based on topicName and partitionId.
+   * @param partitionId the segment partition id on Pinot
+   */
+  public static StreamConfig getStreamConfigFromStreamConfigList(
+      int partitionId, String topicName, List<StreamConfig> streamConfigs) {
+    int rawTopicIndex = getStreamConfigIndexFromPinotPartitionId(partitionId);
+    if (topicName != null && !topicName.isEmpty()) {
+      return streamConfigs.stream().filter(c -> topicName.equals(c.getTopicName()))
```

Comment:
If multiple StreamConfigs match a same topic name, only the first is returned.
we should clarify we don't support multiple "same topic different kafka cluster" in the streamConfigs.